### PR TITLE
Add video quality selection

### PR DESCRIPTION
**Pull request by cemrich**
Implementing #7

### CODE REVIEW

**Comment by userdimi** [comment]
@cemrich I fixed the codeformat and the wording issues. The last thing todo is to eliminate the duplicated tracks in the TrackSelection View. At the moment I display all video tracks that are selected by the TrackSelector. I thing I have to manipulate the array with the video tracks to eliminate the duplicated tracks.

**Comment by cemrich** [comment]
@userdimi Awesome! 

The quality switcher should show up inside the mediathek player, too. It is using the same player (service) so this should be no problem. Also, the menu item should be disabled when only one quality exists.

There are two more features that would work nicely with your implementation, but are not strictly needed:
1. Saving the last selected quality (at least for the current session, or across app restarts)
1. Allow only a maximum quality on metered connections (like Youtube)

Number one is tricky because not all channels or mediathek videos share the same quality settings. Both features should be separate from this pull request.

**Comment by userdimi** [comment]
@cemrich I think that is not a big deal to implement the same quality button in the mediathek player and show it if more then one quality is available. 

Your ideas for the two features are great! I will think about the details of the implementation and give you some feedback.

At the moment I stuck a little by overriding the TrackGroup of the TrackGroupArray. I wrote a detailed question with more information in the Exoplay Issue tracker: https://github.com/google/ExoPlayer/issues/6082

Maybe you have an idea how to solve the issue? I've commited my current changes to my feature branch so feel free to take a look on it.

**Comment by cemrich** [comment]
@userdimi I don't see how you could use a filtered track list on the ```TrackSelectionView```. However, you could provide a ```TrackNameProvider``` (maybe inherit from ```DefaultTrackNameProvider```) and number the duplicate qualities. I don't know if that's possible with the ```TrackNameProvider```s information but it would indicate to the user that duplicate qualities are not a bug but fallback urls.

**Comment by userdimi** [comment]
@cemrich Thank you for the hint. I will take a look at the TrackNameProvider. You are right, it's very uncool to remove the fallback URLs from the TrackSelector 😃

**Comment by userdimi** [comment]
@cemrich I've implemented a CustomTrackNameProvider class. In this class I manipulate the displayed string and set "Fallback" prefix to duplicated tracks. I've implemented the "Quality" menu item in the mediathek player also. Thanks a lot for the great hint with the TrackNameProvider. I've pushed my changes so feel free to check it out. You can change the german wording of the "Fallback" prefix also

File: app/src/main/java/de/christinecoenen/code/zapp/app/player/CustomTrackNameProvider.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	 * @param resources Resources from which to obtain strings.
+	 */
+	CustomTrackNameProvider(Resources resources) {
+		super(resources);
+		this.resources = resources;
+	}
+
+	@Override
+	public String getTrackName(Format format) {
+		String trackName;
```
Comment (cemrich) [inline app/src/main/java/de/christinecoenen/code/zapp/app/player/CustomTrackNameProvider.java]: This class can be heavily simplified by making a call to `super.getTrackName(fomat)`. The resulting string can then be suffixed if needed. All other methods than can be deleted.

**Comment by cemrich** [comment]
Thank you very, very much for your work! 

I've added one last refactoring suggestion. If you don't want to change it, I'll merge inside a feature branch and refactor later :)

File: app/src/main/java/de/christinecoenen/code/zapp/app/player/CustomTrackNameProvider.java (hunk 1 above)
Comment (userdimi) [reply app/src/main/java/de/christinecoenen/code/zapp/app/player/CustomTrackNameProvider.java]: You are totally right. This class could be more simple. I will refractor it tomorrow and will push my changes. Thanks a lot for your feedback.

**Comment by userdimi** [comment]
@cemrich You are welcome. I learned a lot by contributing to your project. I'll looking forward to see this feature in production 🙂

Like a wrote in the reply, will refractor the class and push the changes tomorrow

**Comment by userdimi** [comment]
@cemrich I refactored the CustomTrackNameProvider class and pushed my changes. Thank's a lot for your feedback!

**Pull request merged**